src/platform/silabs/EndpointQueueFilter.cpp

```diff
@@ -0,0 +1,118 @@
+#include "EndpointQueueFilter.h"
+#include <algorithm>
+#include <cctype>
+#include <string.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+namespace chip {
+namespace Inet {
+
+using FilterOutcome = EndpointQueueFilter::FilterOutcome;
+
+namespace {
+
+bool IsValidMdnsHostName(const CharSpan & hostName)
+{
+    for (size_t i = 0; i < hostName.size(); ++i)
+    {
+        char ch_data = *(hostName.data());
```

Comment:
Is there a reason not to do:
```suggestion
    for (char ch_data : hostName)
    {
```
?

Reply:
notice the span interation is not working, will have a check once

Reply:
I don't understand this comment...